SalesforceCommerceCloud/pwa-kit
# [V3][Hooks Integration 🪝] Replace last of the `getProps`

# Description

There were a few places left in the `retail-react-template` where we were still using `getProps`. A) in the `pge-not-found` page and B) in the `_app` component. In this PR I'm replacing those uses of `getProps `with their non-getProps `react-query` equivalent.

_NOTE: Because we use ssrPrepass to get hooks working server-side there was an issue where during prepass the messages aren't available, leading to warnings in the console for missing translations. To work around this I updated the custom error handler to only warn if there are messages defined/loaded and the individual message isn't found._

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Replace _app getProps with useQuery implementation.
- Replace page-not-found getProps with useServerContext
- Remove use of the `withLegacyGetProps` HOC
- Remove any unused tests.
- Fix bug in SDK where it was required that you have an `extraGetPropsArgs` implementation even when you aren't using getProps.
- Remove `getProps` HOC from test render function.

# How to Test-Drive This PR

- Load dev server and set basic features of localization, changing the local, seeing the text change on various pages etc.
- Goto a url where there is no route defined to see that you get a 404 response when the page not found page renders.

# Checklists




## General
